src/source/vector_tile_source.js

```diff
@@ -94,6 +94,24 @@ class VectorTileSource extends Evented implements Source {
         this.load();
     }
 
+    setSourceProperty(name: string, value: mixed) {
+        if (this._tileJSONRequest) {
+            this._tileJSONRequest.cancel();
+        }
+
+        const options = { [name]: value};
+        extend(this, pick(options, ['url', 'scheme', 'tileSize']));
+        this._options = extend({ type: 'vector' }, options);
+
+        if (this.tileSize !== 512) {
+            throw new Error('vector tile sources must have a tileSize of 512');
+        }
+
+        const sourceCache = this.map.style.sourceCaches[this.id];
+        sourceCache.clearTiles();
```

Comment:
The `load()` method should fire the `metadata` and `content` events that trigger a `SourceCache#reload()` and clears tiles when the new tilejson is available. See: 
https://github.com/mapbox/mapbox-gl-js/blob/4eb7d4e58c2fb4483994e7d364016c2188709e1a/src/source/source_cache.js#L69-L83

Is there a reason this additional call to `clearTiles` is needed here?

Reply:
`SourceCache#reload()` doesn't remove existing tiles. It causes that after changing the source URL, you still have cached tiles from previous source URL, because they are not in `errored` state (see https://github.com/mapbox/mapbox-gl-js/blob/master/src/source/source_cache.js#L225).

So I call `clearTiles` to force remove previously cached tiles.

Reply:
We can force `errored` state thought, but it looks not so good as clearing tiles with `clearTiles `